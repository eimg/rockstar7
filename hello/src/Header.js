import React from 'react';

const Header = props => {
	return (
		<h1>Title ({props.count})</h1>
	)
}

export default Header;
