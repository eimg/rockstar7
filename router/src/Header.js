import React from 'react';

import { 
    Link,
    useParams,
} from 'react-router-dom';

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';

const styles = {
    root: {
        flexGrow: 1,
    },
    menuButton: {
        marginRight: 20,
    },
    title: {
        flexGrow: 1,
    },
};

const Header = props => {
    let { label } = useParams();

    return (
        <div style={styles.root}>
            <AppBar position="static">
                <Toolbar>
                    {
                        props.back
                        ? <IconButton 
                            edge="start" 
                            style={styles.menuButton} 
                            color="inherit"
                        >
                            <Link to="/" style={{color: 'white', display: 'inherit'}}>
                                <ArrowBackIcon />
                            </Link>
                        </IconButton>
                        : <IconButton 
                            edge="start" 
                            style={styles.menuButton} 
                            color="inherit"
                            onClick={props.toggleDrawer(true)}
                        >
                            <MenuIcon />
                        </IconButton>
                    }

                    <Typography variant="h6" style={styles.title}>
                        {
                            label
                            ? `${props.title}: ${label}`
                            : props.title
                        }
                    </Typography>
                </Toolbar>
            </AppBar>
        </div>
    );
}

export default Header;
