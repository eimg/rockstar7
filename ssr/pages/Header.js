const Header = props => {
	return (
		<div style={{ padding: 20, textAlign: 'center', background: 'teal', color: 'white' }}>
			<h1>{props.title}</h1>
		</div>
	);
};

export default Header;
