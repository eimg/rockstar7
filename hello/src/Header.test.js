import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

test('Header rendered correctly', () => {
  const tree = renderer
	.create(<Header count={10} />)
	.toJSON();
  expect(tree).toMatchSnapshot();
});
