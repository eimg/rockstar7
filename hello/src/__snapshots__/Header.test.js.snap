// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header rendered correctly 1`] = `
<h1>
  Title (
  10
  )
</h1>
`;
